src/ExpressionEvaluator/CSharp/Source/ExpressionCompiler/Symbols/PlaceholderLocalSymbol.cs

```diff
@@ -45,7 +45,6 @@ internal PlaceholderLocalSymbol(MethodSymbol method, string name, string display
             ReadOnlyCollection<string> tupleElementNames;
             CustomTypeInfo.Decode(alias.CustomTypeInfoId, alias.CustomTypeInfo, out dynamicFlags, out tupleElementNames);
 
-            // Preserve tuple element names. See https://github.com/dotnet/roslyn/issues/13589.
             if (dynamicFlags != null)
             {
                 var flagsBuilder = ArrayBuilder<bool>.GetInstance();
```

Comment:
Unrelated to your change, but could this block (for dynamic case) be factored nicely like you did for tuple case?